### DISCUSSION TITLE
how to manually flash SHA5264 with tasmota if tuyo-connect is not an option?

### DISCUSSION THREAD

**Discussion by deltamelter** [arendst/Tasmota#14789] [Q&A] [answered]
Got this device with a non working firmware to do tuyo-convert, but the device has  test points for 5v, GND, Tx, Rx, RST and GPIO0 
![image](https://user-images.githubusercontent.com/27061628/153227071-c2db18fa-56aa-47e2-b096-1441cc04b05e.png)

What do I need to connect to this to flash manually?  is it a problem that this is 5v and not 3.3v?

These are not my pictures, but from somebody else who has the same device, but was lucky enough to have firmware supporting tuyo-connect
![image](https://user-images.githubusercontent.com/27061628/153227841-59a4815d-68f8-4d5b-8cad-0cd1b6601b8a.png)
https://gist.github.com/en4rab/6bc83b2c9f850ea29eeaa5f1833a21fd

**Comment by sfromis** [comment, marked as answer]
For general wired flashing info, see https://tasmota.github.io/docs/Getting-Started/

While the ESP8285 needs 3.3V, the AMS1117 is a common voltage regulator for making 3.3V out of 5V, Thus I'd expect it to work to provide 5V. Note that common USB/serial adapters may have same voltage on both data and power outputs, in that case you should ideally provide the 5V from another power source, common ground. This avoids 5V getting near the ESP8285 which risks frying the chip. 

The recommended adapter, the "golden CH340G" should be able to provide 5V power (from the USB input) while having 3.3V on the data lones.

**Comment by deltamelter** [comment]
Damnit...   soldered up tiny wires and managed to pull the tx pad off the board.  I don't think there's any chnce of soldering straight to the chip.  can any flash methods still work with only rx?

**Reply by barbudor** [reply]
No, you need TX for the PC to receive the handshakes
Soldering directly on the ESP8266 TX is a bit ... challenging 
![image](https://user-images.githubusercontent.com/2996042/153767001-96da5438-7598-4cf6-9a8e-380a89f43387.png)

**Reply by deltamelter** [reply]
that's what I feared.  I guess I might as well give it a try but will have to find a fine enough wire and tip for the iron.

**Reply by sfromis** [reply]
With a steady hand, you could also try hand-holding a needle to make a connection while doing the flashing. While this is not great for a stable connection, obviously, it should be workable if you're willing to try a few times.

**Comment by deltamelter** [comment]
might work with a needle, thanks for the tip...  I'm using a "golden CH340G" which has a jumper to switch between 3v3 and 5v. Does that need to be on 3v3 for the data lines, or 5v for the 5v pin?  no response so far.  the jumper is set to 3v3

**Reply by sfromis** [reply]
It will have to be on 3.3V for the data lines. Normally, you should use the 3.3V power line for ESP8266, but if you are sure that you need 5V, that's available on that pin too, IIRC it is basically the USB power. Did you make sure to have tx/rx crossed, and to ground gpio0 while applying power?

**Reply by deltamelter** [reply]
the gpio0 to gnd had come loose.. Thanks for the quick replies.  flashed, restarted and working...
Just check out the config etc before desoldering the other pins and powering from mains
Thanks again!!
